Sources/SATSCore/Components/ModernErrorView/ModernErrorView.swift

```diff
@@ -0,0 +1,94 @@
+import SwiftUI
+
+public struct ModernErrorView: View {
+    let viewData: ErrorViewData
+    var onRetry: (() async -> Void)?
+
+    public init(viewData: ErrorViewData, onRetry: (() async -> Void)? = nil) {
+        self.viewData = viewData
+        self.onRetry = onRetry
+    }
+
+    public var body: some View {
+        VStack(spacing: .spacingL) {
+            Spacer()
+
+            icon
+
+            VStack(spacing: .spacingXS) {
+                errorTitle
+                errorMessage
+            }
+
+            Spacer()
+        }
+        .padding(.spacingXL)
+        .lineLimit(nil)
+        .foregroundColor(.onBackgroundPrimary)
+        .overlay { retryButtonContainer }
+        .multilineTextAlignment(.center)
+        .frame(maxWidth: .infinity)
```

Comment:
Can we add .edgesIgnoringSafeArea(.top) for centralizing the content?

Reply:
No, as this depends on the container for the view.

The Demo View is not ignoring the safe area... then the ModernErrorView shouldn't assume